File: Coding Sample/MS.NET/Fundamentals/Persistence/dataiotest2.cs
using System;
using System.IO;

static class Program
{
	public static void Main(string[] args)
	{
		if(args.Length > 2)
		{
			float price = Convert.ToSingle(args[0]);
			short stock = Convert.ToInt16(args[1]);
			string name = args[2].ToUpper();

			StreamWriter writer = new StreamWriter(File.Create("product.txt"));
			writer.WriteLine(price);
			writer.WriteLine(stock);
			writer.WriteLine(name);
			writer.Close();
		}
		else
		{
			StreamReader reader = new StreamReader(File.OpenRead("product.txt"));
			float price = Convert.ToSingle(reader.ReadLine());
			short stock = Convert.ToInt16(reader.ReadLine());
			string name = reader.ReadLine();		
			reader.Close();

			Console.WriteLine($"Price={price}, Stock={stock} and Name={name}");
		}
	}
}

File: Coding Sample/Java/Fundamentals/Persistence/DataIOTest2.java
import java.io.*;

class DataIOTest2{

	public static void main(String[] args) throws Exception{
		if(args.length > 2){
			float price = Float.parseFloat(args[0]);
			short stock = Short.parseShort(args[1]);
			String name = args[2].toUpperCase();
			PrintWriter pw = new PrintWriter(
				new OutputStreamWriter(
				new FileOutputStream("product.txt")));
			pw.println(price);
			pw.println(stock);
			pw.println(name);
			pw.close();
		}else{
			BufferedReader br = new BufferedReader(
				new InputStreamReader(
				new FileInputStream("product.txt")));
			float price = Float.parseFloat(br.readLine());
			short stock = Short.parseShort(br.readLine());
			String name = br.readLine();
			br.close();
			System.out.printf("%s %s %s%n", price, stock, name);
		}
	}
}


